components/datetime/src/scaffold/calendar.rs

```diff
@@ -335,10 +337,7 @@ where
             JapaneseExtended => H::JapaneseExtended::bind(p).load_bound(req),
             Persian => H::Persian::bind(p).load_bound(req),
             Roc => H::Roc::bind(p).load_bound(req),
-            _ => Err(
-                DataError::custom("Don't know how to load data for specified calendar")
-                    .with_debug_context(&self.kind),
-            ),
+            _ => unreachable!(),
```

Comment:
Issue: I won't approve this with a panic. It is not a panic that the compiler can easily eliminate.

Reply:
I fixed this in #6352